src/path-symex/path_symex.cpp

```diff
@@ -290,6 +297,101 @@ void path_symext::symex_malloc(
   assign(state, lhs, rhs);
 }
 
+
+/*******************************************************************\
+
+Function: get_symbol
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+irep_idt get_symbol(const exprt &src, path_symex_statet &state)
+{
+  if(src.id() == ID_typecast)
+    return get_symbol(to_typecast_expr(src).op(), state);
+  else if(src.id() == ID_address_of)
+  {
+    exprt op=to_address_of_expr(src).object();
+
+    if(op.id() == ID_symbol)
+    {
+      var_mapt::var_infot &var_info=state.var_map[op.get(ID_C_full_identifier)];
+
+      if(var_info.full_identifier == op.get(ID_C_full_identifier))
+      {
+        const irep_idt &identifier=to_symbol_expr(op).get_identifier();
+        return identifier;
+      }
+    }
+  }
+
+  return irep_idt();
+}
+
+
+/*******************************************************************\
+
+Function: path_symext::symex_va_arg_next
+
+  Inputs:
+
+ Outputs:
+
+ Purpose:
+
+\*******************************************************************/
+
+void path_symext::symex_va_arg_next(path_symex_statet &state, const exprt &lhs,
+    const side_effect_exprt &code)
+{
+  if(code.operands().size() != 1)
+    throw "va_arg_next expected to have one operand";
+
+  if(lhs.is_nil())
+    return; // ignore
+
+  exprt tmp=code.op0();
+  tmp=state.read(tmp); // to enable constant prop!
+  irep_idt id=get_symbol(tmp, state);
+  exprt rhs=gen_zero(lhs.type());
+
+  if(id != irep_idt())
+  {
+    // strip last name off id to get function name
+    std::size_t pos=id2string(id).rfind("::");
```

Comment:
@kroening there ought to be a better way of doing this, but I'm not sure it actually is possible.